src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -1288,5 +1288,26 @@ public static bool All<TSource>(this ReadOnlySpan<TSource> source, Func<TSource,
             }
             return true;
         }
+
+        public static void CopyPropertiesTo<T, TU>(this T source, TU dest)
+        {
+            var sourceProps = typeof(T).GetProperties().Where(x => x.CanRead).ToList();
+            var destProps = typeof(TU).GetProperties()
+                    .Where(x => x.CanWrite)
+                    .ToList();
```

Comment:
indexer is a property too. do you want to filter out indexers?

`x.GetIndexParameters().Length == 0`?

Reply:
Probably yes, let me add this.

Reply:
Fixed.